### ISSUE TITLE
pip install ctakesclient

### ISSUE THREAD

**Issue by comorbidity** [comorbidity/ctakes-client-python#1]
this package should support "pip install" similar to fhirclient by the same SMART team: 
https://github.com/smart-on-fhir/client-py

Consistency matters. 
First rename this packaging  to be consistent with https://github.com/smart-on-fhir/client-py

**Comment by mikix** [comment]
Here's my proposal:
- Get @comorbidity and @mikix added to the [Machine-Learning-for-Medical-Language](https://github.com/Machine-Learning-for-Medical-Language/) group.
- Move this repo to `Machine-Learning-for-Medical-Language/ctakes-client-py`
- Set up an project page for this on pypi with the name `ctakes`.

Then we have two paths: automatic releases & manual releases.

# Automatic
- Adopt [semantic versioning](https://semver.org/) & [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- Use [GitHub actions](https://guicommits.com/semantic-release-to-automate-versioning-and-publishing-to-pypi-with-github-actions/) to automatically increment version number, tag, and release to pypi on every non-chore commit

So for clarity, what conventional commits & semvar would mean is:
- We add a commit like "fix: don't crash on startup" and then the micro version goes from 1.2.3 to 1.2.4 and pypi gets a new release.
- We add a commit like "feat: add new do_all_the_things API call' and then the minor version goes from 1.2.4. to 1.3.4 and pypi gets a new release.
- We add a breaking change. "feat!: change arguments to do_all_the_things" and then the major version goes from 1.3.4 to 2.3.4.
- We approve a dependabot or similar PR to update a dependency version with the commit "chore: update numpy dependency" and no new release happens. (I don't think that sort of bot setup exists yet, just an example)

# Manual
- We make sure to bump the version number in the source in anticipation of a release.
- We set up a GitHub action for uploading to pypi every time we manually add a new release tag.

# Considerations
Automatic releases would reduce our overhead a touch. The tradeoff being that every commit gets released. Which can feel scary. Mostly a question of how much you trust our tests / how many users we think we have.

I have found that conventional commits do encourage better commit message practice on my part and gets me thinking about my change set a little bit more.

So for that reason, I'm leaning towards automatic, but how do you feel about that @comorbidity? Manual works fine too and is maybe less anxiety-inducing.

**Comment by mikix** [comment]
Oh also a clarification: the title of this ticket mentions `pip install ctakesclient` but the python module we currently install looks like it's called simply `ctakes` -- I'm assuming the title is wrong and we want `pip install ctakes`, but just curious if we should change that before committing to a name.

**Comment by tmills** [comment]
Check email for MLML invites

**Comment by comorbidity** [comment]
> 
@mikix OK for now. There is a ctakes server out there but I'm not sure if it is "pip installable". Perhaps @tmills knows where that Python ctakes server lives? 

@mikix `pip install ctakesclient` is similiar to `pip install fhirclient` so maybe this is best? 
I donno maybe I'm being too picky and we should just use whatever name is best.

**Comment by mikix** [comment]
> pip install ctakesclient is similiar to pip install fhirclient so maybe this is best?
> I donno maybe I'm being too picky and we should just use whatever name is best.

`ctakesclient` is good - doesn't hurt to be a little clearer about intent (plus matching fhirclient). I'll edit my comment above about the plan to include a module rename step.

**Comment by mikix** [comment]
There is an MVP path here with the manual approach. I will take that for now, and we can always layer on conventional commits & auto releasing later, as a separate ticket. I just got excited :P

**Comment by mikix** [comment]
OK, progress update.

Once we land these PRs:
- https://github.com/Machine-Learning-for-Medical-Language/ctakes-client-py/pull/1
- https://github.com/Machine-Learning-for-Medical-Language/ctakes-client-py/pull/3
- (ideally but not required) https://github.com/Machine-Learning-for-Medical-Language/ctakes-client-py/pull/2

Then I can add a github hook for pushing to pypi

**Comment by mikix** [comment]
I updated our pypi publish action to send to the official pypi server and now you can finally do: `pip install ctakesclient`! Woooo!
